tests/test_preview.py

```diff
@@ -0,0 +1,706 @@
+    test_launcher = wlmutils.get_test_launcher()
+    exp = Experiment(exp_name, exp_path=test_dir, launcher=test_launcher)
+
+    db = exp.create_database(port=6780, interface="lo")
+    exp.generate(db, overwrite=True)
+    rs1 = exp.create_run_settings("echo", ["hello", "world"])
+    model = exp.create_model("model_test", run_settings=rs1)
+    # enable key prefixing on ensemble
+    model.enable_key_prefixing()
+    exp.generate(model, overwrite=True)
+
+    preview_manifest = Manifest(db, model)
+
+    # Execute preview method
+    output = previewrenderer.render(exp, preview_manifest)
+
+    # Evaluate output
+    assert "Key prefix" in output
+    assert "model_test" in output
+    assert "Outgoing key collision prevention (key prefixing)" in output
+    assert "Tensors: On" in output
+    assert "DataSets: On" in output
+    assert "Models/Scripts: Off" in output
+    assert "Aggregation Lists: On" in output
+    
+    
+def test_ensembles_preview(test_dir, wlmutils):
+    """
+    Test ensemble preview fields are correct in template render
+    """
+    test_launcher = wlmutils.get_test_launcher()
+    exp = Experiment(
+        "test-ensembles-preview", exp_path=test_dir, launcher=test_launcher
+    )
+
+    # setup ensemble parameter space
+    learning_rate = list(np.linspace(0.01, 0.5))
+    train_params = {"LR": learning_rate}
+
+    # define how each member should run
+    run = exp.create_run_settings(exe="python", exe_args="./train-model.py")
+
+    ensemble = exp.create_ensemble(
+        "Training-Ensemble",
+        params=train_params,
+        params_as_args=["LR"],
+        run_settings=run,
+        perm_strategy="random",
+        n_models=4,
+    )
+
+    preview_manifest = Manifest(ensemble)
+    output = previewrenderer.render(exp, preview_manifest)
+
+    assert "Ensemble name" in output
+    assert "Members" in output
+    assert "Ensemble parameters" in output
+
+
+def test_preview_models_and_ensembles(test_dir, wlmutils):
+    """
+    Test preview of separate model entity and ensemble entity
+    """
+    exp_name = "test-model-and-ensemble"
+    test_dir = pathlib.Path(test_dir) / exp_name
+    test_dir.mkdir(parents=True)
+    test_launcher = wlmutils.get_test_launcher()
+    exp = Experiment(exp_name, exp_path=str(test_dir), launcher=test_launcher)
+
+    rs1 = exp.create_run_settings("echo", ["hello", "world"])
+    rs2 = exp.create_run_settings("echo", ["spam", "eggs"])
+
+    hello_world_model = exp.create_model("echo-hello", run_settings=rs1)
+    spam_eggs_model = exp.create_model("echo-spam", run_settings=rs2)
+    hello_ensemble = exp.create_ensemble("echo-ensemble", run_settings=rs1, replicas=3)
+
+    exp.generate(hello_world_model, spam_eggs_model, hello_ensemble)
+
+    preview_manifest = Manifest(hello_world_model, spam_eggs_model, hello_ensemble)
+    output = previewrenderer.render(exp, preview_manifest)
+
+    assert "Models" in output
+    assert "echo-hello" in output
```

Comment:
nitpic: we should use vars to avoid minor typos or having to update multiple places in the case of test updates. e.g.
```py
hw_name = "echo-hello"
...
assert hw_name in output
```

Reply:
changed to vars in tests